src/marketplace.js

```diff
@@ -0,0 +1,321 @@
+            this.starting = true;
+            await Promise.all([this.db.start(), this.chluIpfs.start()]);
+            this.starting = false;
+            this.started = true;
+            this.events.emit('started');
+        }
+    }
+
+    async stop() {
+        if (this.starting) {
+            throw new Error('Cannot stop Marketplace while it is starting');
+        } else if (this.stopping) {
+            await new Promise(resolve => {
+                this.events.once('stopped', resolve);
+            });
+        } else {
+            this.started = false;
+            this.stopping = true;
+            await Promise.all([this.db.stop(), this.chluIpfs.stop()]);
+            this.stopping = false;
+            this.stopped = true;
+            this.events.emit('stopped');
+        }
+    }
+
+    /**
+     * @typedef {Object} KeyPair
+     * @property {ECPair} keyPair the actual key pair
+     * @property {string} pubKeyMultihash the multihash that can be used to get the PubKey from IPFS
+     * @property {string} source where this comes from. Can be `random`, `memory`, or `fs:(path)`
+     */
+
+    /**
+     * Gets your keypair and related information. If the keypair
+     * was not already loaded, it is loaded from the file system.
+     * If this fails or is not available, then a new key pair is
+     * generated.
+     * 
+     * @memberof Marketplace
+     * @returns {Promise<KeyPair>}
+     */
+    async getKeys() {
+        await this.start();
+        let source = 'memory';
+        if (!this.rootKeyPair) {
+            let fileBuffer = null;
+            if (this.rootKeyPairPath !== false) {
+                fileBuffer = await readFile(this.rootKeyPairPath);
+            }
+            if (fileBuffer) {
+                const wif = fileBuffer.toString('utf-8');
+                this.rootKeyPair = ECPair.fromWIF(wif);
+                source = 'fs:' + this.rootKeyPairPath;
+            } else {
+                this.rootKeyPair = ECPair.makeRandom();
+                if (this.rootKeyPairPath !== false) {
+                    await saveFile(this.rootKeyPairPath, this.rootKeyPair.toWIF());
+                }
+                source = 'random';
+            }
+            const buffer = this.rootKeyPair.getPublicKeyBuffer();
+            this.pubKeyMultihash = await this.chluIpfs.instance.vendor.storePublicKey(buffer);
+            this.chluIpfs.pin(this.pubKeyMultihash);
+            // TODO: request pin?
+        }
+        return {
+            keyPair: this.rootKeyPair,
+            pubKeyMultihash: this.pubKeyMultihash,
+            source
+        }; 
+    }
+
+    /**
+     * Get the list of vendors that is registered in this
+     * marketplace. The ID used to identify each vendor
+     * is the multihash of their public key, which can be
+     * used to get the pub key from IPFS.
+     * 
+     * @returns {Array.<string>}
+     * @memberof Marketplace
+     */
+    async getVendorIDs() {
```

Comment:
Handle Not found case

Reply:
If there are no vendors, it just returns an empty array